sample/main.cpp

```diff
@@ -40,12 +47,21 @@ int main(int argc, char* argv[])
             printHelp(argv[0]);
             return 0;
         }
-        if (arg == "--version" || arg == "-v")
+        else if (arg == "--version" || arg == "-v")
         {
             printVersion(argv[0]);
             return 0;
         }
-        if (arg == "--secret" || arg == "-s")
+        else if (arg == "--port" || arg == "-p")
+        {
+            if (i + 1 == argc)
+            {
+                std::cerr << arg << " required argument - port number.\n";
+                return 1;
+            }
+            port = atoi(argv[++i]);
```

Comment:
`atoi` is awful, it does not check enything. Let's use `std::stoul` or `std::from_chars`.